src/ImageSharp/Formats/Webp/Lossless/Vp8LBackwardRefs.cs

```diff
@@ -1,21 +1,28 @@
 // Copyright (c) Six Labors.
 // Licensed under the Six Labors Split License.
 
+using System.Buffers;
+using SixLabors.ImageSharp.Memory;
+
 namespace SixLabors.ImageSharp.Formats.Webp.Lossless;
 
-internal class Vp8LBackwardRefs
+internal class Vp8LBackwardRefs : IDisposable
 {
-    public Vp8LBackwardRefs(int pixels) => this.Refs = new List<PixOrCopy>(pixels);
+    private readonly IMemoryOwner<PixOrCopy> refs;
+    private int count;
+
+    public Vp8LBackwardRefs(MemoryAllocator memoryAllocator, int pixels)
+    {
+        this.refs = memoryAllocator.Allocate<PixOrCopy>(pixels);
+        this.count = 0;
+    }
+
+    public void Add(PixOrCopy pixOrCopy) => this.refs.Memory.Span[this.count++] = pixOrCopy;
```

Comment:
I imaging resolving the span here is a strong candidate for any slowdown using pooled memory.

Reply:
I'm getting the code right that `Vp8LBackwardRefs` is a temporary object? If yes, we could `Pin` the memory in the ctr, store the handle and the `PixOrCopy*` pointer and `Unpin` the handle in `Dispose`.

Reply:
You are right. The resolving here is quite slow.
That's why I switched from index-based `refs` access to `foreach` in my [second commit](https://github.com/SixLabors/ImageSharp/pull/2940/commits/ff9d3a6223c72ca1c5931593e404ce0070db9d9c).

However, the whole change still makes the encoder almost 5% faster.
Also, the result is more stable and less affected by GC pauses (see the reduced Error value in my first post).
The `refs` are read-mostly, and I don't think that relatively slow `Add` is such a big problem.